packages/components/inputs/select-utils/src/create-select-styles.ts

```diff
@@ -0,0 +1,472 @@
+/*
+  ⚠️ These styles are used for
+    - SelectInput
+    - AsyncSelectInput
+    - CreatableSelectInput
+    - AsyncCreatableSelectInput
+
+  Always check all affected components when making changes here!
+*/
+import { ReactNode } from 'react';
+import {
+  customProperties as vars,
+  designTokens,
+} from '@commercetools-uikit/design-system';
+import { Theme } from '@emotion/react';
+
+type TProps = {
+  isDisabled?: boolean;
+  hasError?: boolean;
+  hasWarning?: boolean;
+  isReadOnly?: boolean;
+  showOptionGroupDivider?: boolean;
+  menuPortalZIndex?: number;
+  iconLeft?: ReactNode;
+  isMulti?: boolean;
+  hasValue?: boolean;
+};
+
+type TBase = {
+  fontColorForInput?: string;
+  borderColorForInput?: string;
+  color?: string;
+  backgroundColor?: string;
+  borderTop?: string;
+  borderColor?: string;
+  boxShadow?: string;
+  pointerEvents?: string;
+};
+
+type TState = {
+  isFocused?: boolean;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+};
```

Comment:
Is there a way to import those from react-select? For example https://github.com/JedWatson/react-select/blob/4e06d22e47bbd82a4f9f67452c7f16787c4ce529/packages/react-select/src/styles.ts#L52-L79

Reply:
For TBase, I tried using `CSSObjectWithLabel` but it caused the build to fail and I got this error.
`The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.` 

For TState, We can use `ClassNamesState` but the implementation is not explicit. So, I'm not sure it is a good idea to use it.